packages/web-component/src/lib/helpers/flowInputs.ts

```diff
@@ -0,0 +1,37 @@
+export const transformFlowInputFormData = (data: string) => {
+  try {
+    const form = JSON.parse(data);
+
+    let formData = form;
+    const vals = Object.values(form);
+
+    // transform values to object structure if needed
+    if (vals.some((s) => typeof s === 'string')) {
```

Comment:
Why do we need this block?
Can't we just do it in the reduce below?

Reply:
I believe we want to duplicate the field's names (e.g. have `email` and `form.email` as well, @Bars92 - fact checking this with @Bars92), so we normalize the structure, otherwise - the `prev` accumulator merges with the original values that don't match the nested structure.

Instead, we can return this :

```
Object.fromEntries(
      Object.keys(form)
        .map((key) =>
          typeof form[key] !== 'string'
            ? [key, form[key]]
            : [key, { value: form[key] }]
        ).flatMap(([name, value]) => ([
          [name, value],
          [`form.${name}`, value]
        ])))
```

WDYT @nirgur?

Reply:
@nirgur Ok, validated with @Bars92 - we do need the duplicated field names. Pushed the above fix.